ui/raidboss/data/07-dt/trial/zelenia-ex.ts

```diff
@@ -1,14 +1,638 @@
+            break;
+          case 'AA18': // Roseblood: 6th Bloom
+            data.phase = 'bloom6';
+            break;
+        }
+      },
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait Collect',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'] },
+      preRun: (data, matches) =>
+        data.escelonFallBaits.push(matches.count === '2F6' ? 'near' : 'far'),
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait Cleanup',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'], capture: false },
+      delaySeconds: 30,
+      suppressSeconds: 30,
+      run: (data) => data.escelonFallBaits = [],
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'], capture: false },
+      durationSeconds: 19,
+      suppressSeconds: (data) => data.escelonFallBaits.length > 1 ? 20 : 0,
+      infoText: (data, _matches, output) => {
+        if (data.escelonFallBaits.length !== 2)
+          return;
+        const [bait1, bait2] = data.escelonFallBaits;
+        if (bait1 === undefined || bait2 === undefined)
+          return;
+
+        if (bait1 === bait2) {
+          return output.swapAfterFirst!({
+            first: output[bait1]!(),
+          });
+        }
+        return output.swapAfterSecond!({
+          first: output[bait1]!(),
+        });
+      },
+      outputStrings: {
+        near: {
+          en: 'Near',
+        },
+        far: {
+          en: 'Far',
+        },
+        swapAfterFirst: {
+          en: '${first} bait first, Swap after first+third',
+        },
+        swapAfterSecond: {
+          en: '${first} bait first, Swap after second',
+        },
+      },
+    },
+    {
+      id: 'ZeleniaEx Thorned Catharsis',
+      type: 'StartsUsing',
+      netRegex: { id: 'A89E', source: 'Zelenia', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'ZeleniaEx Shock P1 Tower',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.shockSpread, capture: true },
+      condition: (data, matches) =>
+        conditions.targetIsYou()(data, matches) && data.phase === 'phase1',
+      infoText: (_data, _matches, output) => output.tower!(),
+      outputStrings: {
+        tower: {
+          en: 'Donut on you, get tower',
+        },
+      },
+    },
+    {
+      id: 'ZeleniaEx Shock Spread',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.shockSpread, capture: true },
+      condition: conditions.targetIsYou(),
+      response: Responses.spread(),
+    },
+    {
+      id: 'ZeleniaEx Shock Spread Move Reminder',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.shockSpread, capture: true },
+      condition: conditions.targetIsYou(),
+      delaySeconds: 7,
+      response: Responses.moveAway(),
+    },
```

Comment:
If `headMarkerData.shockSpread` is the donut headmarker per `ZeleniaEx Shock P1 Tower`, should this trigger have a phase filter to exclude phase1?   I think the donut players remain in the 'donut hole'?

Reply:
`shockSpread` headmarkers only happen once in the fight, during p1.

`shockDonut` happens twice (once during p1, once during Escelon 2).